riscv/csrs.cc

```diff
@@ -1347,3 +1347,54 @@ void sstateen_csr_t::verify_permissions(insn_t insn, bool write) const {
 
   hstateen_csr_t::verify_permissions(insn, write);
 }
+
+// implement class fcsr_csr_t
+fcsr_csr_t::fcsr_csr_t(processor_t* const proc, const reg_t addr, csr_t_p upper_csr, csr_t_p lower_csr, const unsigned upper_lsb):
+  composite_csr_t(proc, addr, upper_csr, lower_csr, upper_lsb) {
+}
+
+void fcsr_csr_t::verify_permissions(insn_t insn, bool write) const {
+  if (proc->extension_enabled(EXT_SMSTATEEN) && proc->extension_enabled(EXT_ZFINX)) {
```

Comment:
Commit 402fccb0747334fabac6ad152192765543e27f7b needs to be broken up because it's not clear where the behavioral changes are. First, implement `fcsr_csr_t`, `senvcfg_t`, and `henvcfg_t` as blank placeholder classes with no behavior changes. Then add the Smstateen changes as a separate commit so it's clear what's new.

Reply:
OK.